File: app/src/main/java/vn/loitp/app/activity/customviews/seekbar/boxedverticalseekbar/BoxedVerticalSeekBarActivity.java
package vn.loitp.app.activity.customviews.seekbar.boxedverticalseekbar;

import android.graphics.Color;
import android.os.Bundle;
import android.widget.TextView;

import com.core.base.BaseFontActivity;
import com.core.utilities.LLog;

import java.util.ArrayList;
import java.util.List;

import loitp.basemaster.R;
import vn.loitp.views.seekbar.boxedverticalseekbar.BoxedVertical;

//https://github.com/alpbak/BoxedVerticalSeekBar?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=6291
public class BoxedVerticalSeekBarActivity extends BaseFontActivity {
    private TextView tv;
    private List<String> stringList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        BoxedVertical bv = (BoxedVertical) findViewById(R.id.boxed_vertical);
        tv = (TextView) findViewById(R.id.tv);

        bv.setOnBoxedPointsChangeListener(new BoxedVertical.OnValuesChangeListener() {
            @Override
            public void onPointsChanged(BoxedVertical boxedPoints, final int value) {
                LLog.INSTANCE.d(TAG, "onPointsChanged " + value);
                stringList.add(0, "onPointsChanged " + value);
                print();
            }

            @Override
            public void onStartTrackingTouch(BoxedVertical boxedPoints) {
                LLog.INSTANCE.d(TAG, "onStartTrackingTouch");
                stringList.add(0, "onStartTrackingTouch");
                print();
            }

            @Override
            public void onStopTrackingTouch(BoxedVertical boxedPoints) {
                LLog.INSTANCE.d(TAG, "onStopTrackingTouch");
                stringList.add(0, "onStopTrackingTouch");
                print();
                bv.setBackgroundColor(Color.RED);
                bv.setProgressColor(Color.GREEN);
            }
        });
    }

    private String x;

    private void print() {
        if (stringList.size() > 30) {
            stringList.remove(stringList.size() - 1);
        }
        String x = "";
        for (String s : stringList) {
            x += "\n" + s;
        }
        tv.setText(x);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_boxed_vertical_seekbar;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/seekbar/boxedverticalseekbar/BoxedVerticalSeekBarActivity.kt
package vn.loitp.app.activity.customviews.seekbar.boxedverticalseekbar

import abak.tr.com.boxedverticalseekbar.BoxedVertical
import android.graphics.Color
import android.os.Bundle
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import kotlinx.android.synthetic.main.activity_seekbar_boxed_vertical.*
import vn.loitp.app.R
import java.util.* // ktlint-disable no-wildcard-imports

// https://github.com/alpbak/BoxedVerticalSeekBar?utm_source=android-arsenal.com&utm_medium=referral&utm_campaign=6291
@LogTag("BoxedVerticalSeekBarActivity")
@IsFullScreen(false)
class BoxedVerticalSeekBarActivity : BaseFontActivity() {
    private val stringList = ArrayList<String>()

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_seekbar_boxed_vertical
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        boxedVertical.setOnBoxedPointsChangeListener(object : BoxedVertical.OnValuesChangeListener {
            override fun onPointsChanged(boxedPoints: BoxedVertical, value: Int) {
                logD("onPointsChanged $value")
                stringList.add(index = 0, element = "onPointsChanged $value")
                print()
            }

            override fun onStartTrackingTouch(boxedPoints: BoxedVertical) {
                logD("onStartTrackingTouch")
                stringList.add(index = 0, element = "onStartTrackingTouch")
                print()
            }

            override fun onStopTrackingTouch(boxedPoints: BoxedVertical) {
                logD("onStopTrackingTouch")
                stringList.add(index = 0, element = "onStopTrackingTouch")
                print()
                boxedVertical.setBackgroundColor(Color.RED)
            }
        })
    }

    private fun print() {
        if (stringList.size > 30) {
            stringList.removeAt(stringList.size - 1)
        }
        var x = ""
        for (s in stringList) {
            x += "\n" + s
        }
        textView?.text = x
    }
}
